src/coord/src/coord.rs

```diff
@@ -742,7 +748,8 @@ where
                 biased;
 
                 Some(m) = internal_cmd_rx.recv() => m,
-                Some(m) = self.dataflow_client.recv() => Message::Worker(m),
+                Some(m) = self.storage_client.recv() => Message::Worker(m),
+                Some(m) = self.compute_client.recv() => Message::Worker(m),
```

Comment:
because this is `biased;` it will always pull storage messages first, if the channel has some. Maybe the bug is this needs to be re-ordered, or we need a sub-`select!` so that the previous behavior (random-ish) happens only after the biased `internal` commands

Reply:
This is a good point. Related experiences indicate that there are a pile of other bugs that would prevent this from working too; no sweat about reviewing or understanding it yet!